tests/parallel/test-http-client-max-header-size_default.js

```diff
@@ -0,0 +1,40 @@
+'use strict';
+require('../common');
+
+var assert = require('assert');
+var http = require('http');
+
+var s = "x".repeat(1024 * 1024);
+
+var header = {
+  'x-custom-header': s,
+  'Connection':'keep-alive',
+  'Content-Length': 0,
+  'Content-Type':'application/json; charset=UTF-8'
+};
+
+var server = http.createServer(function(req, res) {
+  // code should not get here
+  assert(false);
+
+  res.end();
+
+  server.close();
+});
+
+server.listen(0, function() {
+  var options = {
+    port: server.address().port,
+    method: 'GET',
+    headers: header
+  };
+
+  var r = http.request(options);
+  r.on('error', function(err) {
```

Comment:
If no error occurs, this test will still pass!  I think you want to add something like:
```javascript
var had_error;
r.on('error', function(err) { had_error = true; });
process.on('exit', function() { assert(had_error); });
```